src/Solution/ParticleTracker/Method.f90

```diff
@@ -151,6 +152,38 @@ subroutine pass(this, particle)
     call pstop(1, "pass must be overridden")
   end subroutine pass
 
+  !> @brief Save a particle's current state.
```

Comment:
@aprovost-usgs I factored out a `save()` routine here to handle the special treatment for period/timestep in case of times right on the boundary between time steps, per offline discussion